### ISSUE TITLE
Webpack 5.10.0 build failures

### ISSUE THREAD

**Issue by haacked** [ashmind/mirrorsharp#142]
Hello! I upgraded my local project to webpack 5.10.0 and got the following build failures:

<details>
<code><pre>
ERROR in ./node_modules/mirrorsharp/main/editor.js 52:0-37
Module not found: Error: Can't resolve 'codemirror/mode/clike/clike' in '/Users/haacked/dev/aseriousbiz/abbot/src/Abbot.Web/node_modules/mirrorsharp/main'
Did you mean 'clike.js'?
BREAKING CHANGE: The request 'codemirror/mode/clike/clike' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ./node_modules/mirrorsharp/mirrorsharp.js 3:0-42 7:19-25
 @ ./assets/js/editor/code.js 2:0-38 96:18-29
 @ ./assets/index.js 12:0-42 63:4-14

ERROR in ./node_modules/mirrorsharp/main/hinter.js 21:0-41
Module not found: Error: Can't resolve 'codemirror/addon/hint/show-hint' in '/Users/haacked/dev/aseriousbiz/abbot/src/Abbot.Web/node_modules/mirrorsharp/main'
Did you mean 'show-hint.js'?
BREAKING CHANGE: The request 'codemirror/addon/hint/show-hint' failed to resolve only because it was resolved as fully specified
(probably because the origin is a '*.mjs' file or a '*.js' file where the package.json contains '"type": "module"').
The extension in the request is mandatory for it to be fully specified.
Add the extension to the request.
 @ ./node_modules/mirrorsharp/main/editor.js 56:0-37 560:31-37
 @ ./node_modules/mirrorsharp/mirrorsharp.js 3:0-42 7:19-25
 @ ./assets/js/editor/code.js 2:0-38 96:18-29
 @ ./assets/index.js 12:0-42 63:4-14
</pre></code>
</details>

There appears to be two possible fixes.

1. Remove the `"type": "module"` from the output `package.json`. I imagine this may cause other issues.
2. Use the file extension on the following two lines:

https://github.com/ashmind/mirrorsharp/blob/78c4e57471c7ee0c7ea1d9e39ee1910da705b17e/WebAssets/ts/main/editor.ts#L2
Becomes
```js
import 'codemirror/mode/clike/clike.js';
```

And
https://github.com/ashmind/mirrorsharp/blob/dfa18f7ec1e4a4d168aa58c616614d87e31d0001/WebAssets/ts/main/hinter.ts#L2
becomes
```js
import 'codemirror/addon/hint/show-hint.js';
```

Those changes seem to fix things on my machine.

**Comment by haacked** [comment]
This seems to be a breaking change in `webpack@5.0.0`. I tried using that version and got the same error. The issue goes away if I downgrade back to `webpack@4.44.2`.

**Comment by ashmind** [comment]
Very interesting. I have jumped to modules a bit early I think, before everything supports them consistently.
Longer term I do expect it to work without extension in both node and webpack, but there might be some rough edges in both -- I'll investigate.

**Comment by ashmind** [comment]
Somehow I assumed it was complicated and postponed it, but it was actually very simple and literally same fix as you suggested.  
Anyways, should be fixed in [2.0.4](https://www.npmjs.com/package/mirrorsharp/v/2.0.4) -- pls reopen if it does not help.
